drivers/dma/dma_mcux_lpc.c

```diff
@@ -191,12 +202,75 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+		LOG_ERR("DMA attempted to configure an Otrig mux"
+		" while the mux was in use.");
+		return -EACCES;
+	}
+
+	if (config->source_chaining_en || config->dest_chaining_en) {
+		/* Chaining is enabled */
+		LOG_DBG("link dma 0 channel %d with channel %d",
+			channel, config->linked_channel);
+		uint8_t is_otrig_available = 0;
+
+		for (otrig_index = 0; otrig_index < dev_config->num_of_otrigs;
+			++otrig_index) {
+			if (dma_data->otrig_array[otrig_index].linked_channel == -1 ||
+			    dma_data->otrig_array[otrig_index].source_channel == channel) {
+				if (dma_data->otrig_array[otrig_index].source_channel == channel) {
+					DMA_DisableChannel(DEV_BASE(dev),
+						dma_data->otrig_array[otrig_index].linked_channel);
+					dma_data->num_channels_used--;
+				}
+				is_otrig_available = 1;
+				break;
+			}
+		}
+		if (!is_otrig_available) {
+			LOG_ERR("Calling function tried to setup up multiple"
+			" channels to be configured but the dma driver has"
+			" run out of OTrig Muxes");
+			k_sem_give(&is_otrig_being_configured);
+			return -EINVAL;
+		}
+
+		/* Since INPUTMUX handles the dma signals and
+		 * must be hardware triggered via the INPUTMUX
+		 * hardware.
+		 */
+		DEV_BASE(dev)->CHANNEL[config->linked_channel].CFG |=
+			DMA_CHANNEL_CFG_HWTRIGEN_MASK;
+
+		DMA_EnableChannel(DEV_BASE(dev), config->linked_channel);
+
+		/* Link OTrig Muxes with passed-in channels */
+		INPUTMUX_AttachSignal(INPUTMUX, otrig_index,
+			dev_config->otrig_base_address);
+		INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel,
+				dev_config->itrig_base_address);
```

Comment:
It seems this code will only work for channel 0, linked via OTRIG0.

The first call to `INPUTMUX_AttachSignal` will result in `0x0` being written to `0x160 + (otrig_index * 4)`. This will configure the OTRIG at `otrig_index` to use channel 0 as an input, as expected. *However*, if another channel is selected as the software-triggered channel to be linked, then the selected channel would still be zero, as the value written to the OTRIG inputmux register is what determines which channel is used as the input.

To resolve this, it seems you need to use following function call: `INPUTMUX_AttachSignal(INPUTMUX, otrig_index, dev_config->otrig_base_address + channel)`

The second call to `INPUTMUX_AttachSignal` will result in `0xF` being written to `0xE0 + (config->linked_channel * 4)`. This will configure channel `config->linked_channel` to use `OTRIG0` as an input. *However*, if `otrig_index != 0`, `ORIG0 ` would still be selected, which would be incorrect.

To resolve this, it seems you need to use following function call: `INPUTMUX_AttachSignal(INPUTMUX, config->linked_channel, dev_config->itrig_base_address + otrig+index);`

Reply:
This has been changed thank you.